cmd/juju/commands/bootstrap.go

```diff
@@ -652,48 +649,19 @@ func (c *bootstrapCommand) setBootstrapEndpointAddress(
 	}
 	bootstrapInstance := instances[0]
 
-	cfg := environ.Config()
-	info, err := legacyStore.ReadInfo(
-		configstore.EnvironInfoName(c.controllerName, cfg.Name()),
-	)
-	if err != nil {
-		return errors.Annotate(err, "failed to get connection info")
-	}
-
 	// Don't use c.ConnectionEndpoint as it attempts to contact the state
 	// server if no addresses are found in connection info.
-	endpoint := info.APIEndpoint()
 	netAddrs, err := bootstrapInstance.Addresses()
 	if err != nil {
 		return errors.Annotate(err, "failed to get bootstrap instance addresses")
 	}
+	cfg := environ.Config()
 	apiPort := cfg.APIPort()
 	apiHostPorts := network.AddressesWithPort(netAddrs, apiPort)
-	addrs, hosts, addrsChanged := prepareEndpointsForCaching(
-		info, [][]network.HostPort{apiHostPorts}, network.HostPort{},
-	)
-	if !addrsChanged {
-		// Something's wrong we already have cached addresses?
-		return errors.Annotate(err, "cached API endpoints unexpectedly exist")
-	}
-	endpoint.Addresses = addrs
-	endpoint.Hostnames = hosts
-	info.SetAPIEndpoint(endpoint)
-	err = info.Write()
-	if err != nil {
-		return errors.Annotate(err, "failed to write API endpoint to connection info")
-	}
 
-	controllerStore := c.ClientStore()
-	err = controllerStore.UpdateController(c.controllerName, jujuclient.ControllerDetails{
-		hosts,
-		endpoint.ServerUUID,
-		addrs,
-		endpoint.CACert,
-	})
-	if err != nil {
-		return errors.Trace(err)
+	params := juju.ControllerUpdateParams{
+		ControllerName: c.controllerName,
 	}
-
-	return nil
+	fullModelName := configstore.EnvironInfoName(c.controllerName, cfg.Name())
```

Comment:
It would be nice not to have to pass the model name in, since it's not needed by the new stuff.

Would it be a pain to do the following in UpdateControllerAddresses instead?
 - get the controller EnvironInfo (using configstore.AdminModelName for model name), and identify its UUID
 - list all EnvironInfos with controller UUID equal to above
 - update all of their addresses